opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java

```diff
@@ -147,21 +171,149 @@ public void run() {
 
+    protected Map<String, Object> createMapping(
+            Map<Integer, Map<String, Object>> alarmMappings,
+            NorthboundAlarm alarm) {
+        Map<String, Object> mapping;
+        mapping = new HashMap<String, Object>();
+        mapping.put("ackUser", alarm.getAckUser());
+        mapping.put("appDn", alarm.getAppDn());
+        mapping.put("logMsg", alarm.getLogMsg());
+        mapping.put("objectInstance", alarm.getObjectInstance());
+        mapping.put("objectType", alarm.getObjectType());
+        mapping.put("ossKey", alarm.getOssKey());
+        mapping.put("ossState", alarm.getOssState());
+        mapping.put("ticketId", alarm.getTicketId());
+        mapping.put("alarmUei", alarm.getUei());
+        mapping.put("ackTime", nullSafeToString(alarm.getAckTime(), ""));
+
+        AlarmType alarmType = alarm.getAlarmType() == null ? AlarmType.NOTIFICATION
+                                                          : alarm.getAlarmType();
+        mapping.put("alarmType", alarmType.name());
+
+        String count = alarm.getCount() == null ? "1"
+                                               : alarm.getCount().toString();
+        mapping.put("count", count);
+
+        mapping.put("firstOccurrence",
+                    nullSafeToString(alarm.getFirstOccurrence(), ""));
+        mapping.put("alarmId", alarm.getId().toString());
+        mapping.put("ipAddr", nullSafeToString(alarm.getIpAddr(), ""));
+        mapping.put("lastOccurrence",
+                    nullSafeToString(alarm.getLastOccurrence(), ""));
+
+        if (alarm.getNodeId() != null) {
+            LOG.debug("Adding nodeId: " + alarm.getNodeId().toString());
+            mapping.put("nodeId", alarm.getNodeId().toString());
+            String nodeLabel = m_nodeDao.getLabelForId(alarm.getNodeId());
+            mapping.put("nodeLabel", nodeLabel == null ? "?" : nodeLabel);
+        } else {
+            mapping.put("nodeId", "");
+            mapping.put("nodeLabel", "");
+        }
+
+        String poller = alarm.getPoller() == null ? "localhost"
+                                                 : alarm.getPoller().getId();
+        mapping.put("distPoller", poller);
+
+        String service = alarm.getService() == null ? "" : alarm.getService();
+        mapping.put("ifService", service);
+
+        mapping.put("severity", nullSafeToString(alarm.getSeverity(), ""));
+        mapping.put("ticketState",
+                    nullSafeToString(alarm.getTicketState(), ""));
+
+        mapping.put("x733AlarmType", alarm.getX733Type());
+
+        try {
+            mapping.put("x733ProbableCause",
+                        nullSafeToString(x733ProbableCause.get(alarm.getX733Cause()),
+                                         ""));
+        } catch (Exception e) {
+            LOG.info("Exception caught setting X733 Cause: {}",
+                     alarm.getX733Cause(), e);
+            mapping.put("x733ProbableCause", "");
```

Comment:
If we were not able to map this, shouldn't we set something like "ERROR" or "UNKNOWN"?